src/ImmersedBoundaries/mask_immersed_field.jl

```diff
@@ -49,6 +68,18 @@ end
     @inbounds field[i, j, k] = scalar_mask(i, j, k, grid, grid.immersed_boundary, loc..., value, field, mask)
 end
 
+function mask_immersed_field_xy!(sumofarrays::SumOfArrays, grid::ImmersedBoundaryGrid, loc, value; k, mask)
+    arch = @inbounds architecture(sumofarrays.arrays[1])
+    loc = instantiate.(loc)
```

Comment:
```suggestion
    loc = map(instantiate, loc)
```

broadcasting takes longer to compile and can lead to type inference issues